### PR TITLE
implement RadixSort

### PR DESCRIPTION
**Pull request by EAlexa**
resolve issue #5

### CODE REVIEW

File: src/main/java/com/williamfiset/algorithms/sorting/RadixSort.java [hunk 1]
```diff
@@ -1,8 +1,44 @@
...
+      }
+    }
+    return max;
+  }
+
   public static void radixSort(int[] ar) {
-    // TODO: https://github.com/williamfiset/Algorithms/issues/5
+    int bucket[][] = new int[10][10];
+    int bucket_count[] = new int[10];
+    int i, j, k, remainder, numDig = 0, divisor = 1, largest, passNo;
```
Comment (williamfiset) [inline src/main/java/com/williamfiset/algorithms/sorting/RadixSort.java:18]: Please reduce the scope of all these variables as much as possible to make the code easier to follow? This is how bugs happen.

File: src/main/java/com/williamfiset/algorithms/sorting/RadixSort.java [hunk 2]
```diff
@@ -1,8 +1,44 @@
-/** NOTE: This file is still under development */
 package com.williamfiset.algorithms.sorting;
 
 public class RadixSort {
+
+  public static int getMax(int[] array) {
```
Comment (williamfiset) [inline src/main/java/com/williamfiset/algorithms/sorting/RadixSort.java:5]: Make private

File: src/main/java/com/williamfiset/algorithms/sorting/RadixSort.java [hunk 3]
```diff
@@ -1,8 +1,44 @@
-/** NOTE: This file is still under development */
 package com.williamfiset.algorithms.sorting;
```
Comment (williamfiset) [inline src/main/java/com/williamfiset/algorithms/sorting/RadixSort.java:2]: Can you add a description of the algorithm, how it works, what type of data you want to sort with it and some time complexity informaiton?

File: src/test/java/com/williamfiset/algorithms/sorting/RadixSortTest.java [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+  }
+
+  @Test
+  public void testRadixSort() {
+    int[] array = {90, 23, 101, 45, 65, 23, 67, 89, 34, 23};
+    RadixSort.radixSort(array);
+    assertEquals(23, array[0]);
+    assertEquals(101, array[array.length - 1]);
+    assertEquals(34, array[3]);
+  }
```
Comment (williamfiset) [inline src/test/java/com/williamfiset/algorithms/sorting/RadixSortTest.java:26]: Can Radix sort handle negative numbers? Add a tet to verify the expected functionality when that happens.

File: src/test/java/com/williamfiset/algorithms/sorting/RadixSortTest.java [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+  @Test
+  public void testGetMax() {
+    int[] array = {3, 90, 112, 56};
+    int[] array2 = {5, 9, 1, 42, 117};
+    int max = RadixSort.getMax(array);
+    int max2 = RadixSort.getMax(array2);
+    assertEquals(112, max);
+    assertEquals(117, max2);
+  }
+
```
Comment (williamfiset) [inline src/test/java/com/williamfiset/algorithms/sorting/RadixSortTest.java:18]: Can you add more test coverage? Can you also add some randomized tests? E.g:

https://github.com/williamfiset/Algorithms/blob/master/src/main/java/com/williamfiset/algorithms/sorting/SelectionSort.java#L45

File: src/test/java/com/williamfiset/algorithms/sorting/RadixSortTest.java (hunk 4 above)
Comment (EAlexa) [reply src/test/java/com/williamfiset/algorithms/sorting/RadixSortTest.java:26]: This implementation can't handle negative numbers

**Comment by EAlexa** [comment]
I am closing this, algorithm does not work as expected.
Will start a new PR soon

File: src/test/java/com/williamfiset/algorithms/sorting/RadixSortTest.java (hunk 4 above)
Comment (williamfiset) [reply src/test/java/com/williamfiset/algorithms/sorting/RadixSortTest.java:26]: It's ok if it doesn't, I don't think Radix sort does in general. I just care that it's documented :)

**Pull request closed without merging**